src/sql/services/scripting/scriptingService.ts

```diff
@@ -99,6 +104,11 @@ export class ScriptingService implements IScriptingService {
 		this._providers[providerId] = provider;
 	}
 
+	public isProviderRegistered(providerId: string): boolean {
+		let provider = this._providers[providerId];
+		return !!provider;
```

Comment:
doesn't having a return type of `boolean` automatically do the `!!` conversion?

Reply:
No - it says it "should" be a boolean, not that it is. It'd just return a reference to the provider, which would let us do a falsy check by not have a boolean. Therefore it's nicer to have it really return an actual boolean instead.